misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -209,7 +210,15 @@ internal class RealTransacter private constructor(
     }
 
     override fun <T : DbEntity<T>> load(id: Id<T>, type: KClass<T>): T {
-      return session.get(type.java, id)
+      try {
+        return session.get(type.java, id)
+      } catch (e : TypeMismatchException) {
+        throw InvalidChildEntityLoadException(e)
+      }
+    }
+
+    override fun  <R : DbRoot<R>, T : DbSharded<R, T>> loadByGid(gid: Gid<R, T>, type: KClass<T>): T {
```

Comment:
Love the idea but not sold on the name. It's a bit long for something that is going to be one of the most commonly used methods. `loadSharded` rolls a bit easier on the tongue but is still long on the screen. Damn Java and its type erasure, otherwise we could have overloaded the method. :-( I'm almost tempted to rename the current `load` method to `loadUnsharded` and make this the short `load` method. This will be more common. Anyway we can iterate on the name.

Reply:
I'm going to use `loadSharded` for now - For a little while (in development) I had it working by overloading `load`, i'm unsure what Java's type erasure is, or how that would've surfaced a problem

Reply:
Oh yeah that's true. `Gid` won't be erased so it should work by overloading. I would suggest you overload `load`!